src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -402,7 +402,14 @@ private void updateScore(Result result, Set<ProgrammingExerciseTestCase> success
             }
 
             // The score is calculated as a percentage of the maximum points
-            long score = Math.round(successfulTestPoints / maxScoreRespectingZeroPointExercises * 100.0);
+            long score;
+            if (maxScoreRespectingZeroPointExercises > 0) {
+                score = Math.round(successfulTestPoints / maxScoreRespectingZeroPointExercises * 100.0);
+            }
+            else {
+                // special case when exercise has no regular points, but bonus points (e.g. 0 points, 5 bonus points -> score between 100% and 105%)
+                score = Math.round((PLACEHOLDER_POINTS_FOR_ZERO_POINT_EXERCISES + successfulTestPoints) / PLACEHOLDER_POINTS_FOR_ZERO_POINT_EXERCISES * 100.0);
```

Comment:
This seems very weird to me. Could we discuss that again? I think this is a bad way to handle pure bonus exercises because the score should still reflect the percentage of the bonus points. I am also wondering how that influences the computation of the course-total points.

Reply:
Yes, the statement in the comment is wrong. Basically, percentages do not make much sense here, because the base value (points) is 0. I think we need to solve this case differently.

Maybe it would be better to split the score for normal points and the score for bonus points in the future to support this case in a better way.

An we need to make sure the course-total points and the export in the course scores are correct

Reply:
As discussed in the meeting, we are allowing a max score of 100% as maxPoints are the amount of bonus points.
e.g. exercise with 0 normal points 5 bonus points -> 2 tests (each giving 1 bonus point) correct -> 40% score as 2/5.

Reply:
Tests concerning the course-score points can be done in a follow up.